src/settings/NoticePolicy/components/DetailSections/components/NoticeCard/NoticeCard.test.js

```diff
@@ -0,0 +1,287 @@
+  if (!value) {
+    it(`should render ${label}`, () => {
+      const testIdValue = screen.getByTestId(testId);
+
+      expect(testIdValue).toBeVisible();
+      expect(within(testIdValue).getByText(`ui-circulation.settings.noticePolicy.notices.${label}`)).toBeVisible();
+    });
+  } else {
+    it(`should render ${label} with ${value} value`, () => {
+      const testIdValue = screen.getByTestId(testId);
+
+      expect(testIdValue).toBeVisible();
+      expect(within(testIdValue).getByText(value)).toBeVisible();
+      expect(within(testIdValue).getByText(`ui-circulation.settings.noticePolicy.notices.${label}`)).toBeVisible();
+    });
+  }
+};
+const testNotShownValues = (testId) => {
+  it(`should not render ${testId} element`, () => {
+    expect(screen.queryByTestId(document.documentElement, testId)).not.toBeInTheDocument();
+  });
+};
+
+describe('View NoticeCard in DetailSection', () => {
+  const noticeCardPropsCheckout = {
+    index: 0,
+    intl: { formatMessage: jest.fn() },
+    notice: {
+      format: 'Email',
+      isRecurring: jest.fn(() => true),
+      realTime: false,
+      sendOptions: {
+        isBeforeOrAfter: jest.fn(() => true),
+        isFrequencyAvailable: jest.fn(() => true),
+        isLoanDueDateTimeSelected: jest.fn(() => true),
+        isSendOptionsAvailable: jest.fn(() => true),
+        sendWhen: 'Check out',
+      },
+      templateId: 'testTemplateId',
+    },
+    sendEventTriggeringIds: [
+      'Due date',
+      'Aged to lost',
+    ],
+    sendEvents: [...noticesSendEvents],
+    templates: [{
+      label: 'TestLabel2',
+      value: 'testTemplateId',
+    }],
+    triggeringEvents: [...loanNoticesTriggeringEvents],
+  };
+  const noticeCard = {
+    header:  'ui-circulation.settings.noticePolicy.countableNotice',
+  };
+
+  describe('with default props and send notice when "check out"', () => {
+    beforeEach(() => {
+      render(<NoticeCard {...noticeCardPropsCheckout} />);
+    });
+
+    it('should render "NoticeCard" component', () => {
+      expect(screen.getByTestId('noticeCard')).toBeVisible();
+    });
+
+    it('should render correct header', () => {
+      expect(within(screen.getByTestId('header')).getByText(noticeCard.header)).toBeVisible();
+    });
+
+    describe('test KeyValues', () => {
+      noticeCardPropsCheckout.templates.map((template) => {
+        return testKeyValues('noticeCardTemplate', 'template', template.label);
+      });
+
+      testKeyValues('noticeCardViaText', 'via');
+      testKeyValues(
+        'noticeCardFormat',
+        'format',
+        `ui-circulation.settings.noticePolicy.notices.${noticeCardPropsCheckout.notice.format.toLowerCase()}`
+      );
+      testKeyValues(
+        'noticeCardTriggeringEvent',
+        'triggeringEvent',
+        noticeCardPropsCheckout.triggeringEvents[1].label
+      );
+
+      describe('when "sendOptions" is available', () => {
+        testKeyValues('noticeSend', 'send');
```

Comment:
i think it will be better move test ids into separate map, especially which are checked for absent in the document.